### PR TITLE
Add groups to programCache and fix observations updates in obsList

### PR DESCRIPTION
**Pull request by hugo-vrijswijk**
- Move program groups query to programCache
- Fix new observations not appearing due to groupElements not being updated

Moving items between groups is not properly updated yet, this needs odb support

### CODE REVIEW

File: common/src/main/scala/explore/cache/ProgramCache.scala [hunk 1]
```diff
@@ -20,13 +23,19 @@ import lucuma.core.model.Observation
 import lucuma.core.model.Program
 import lucuma.core.model.Target
 import lucuma.schemas.ObservationDB
+import lucuma.schemas.ObservationDB.Enums.EditType
+import lucuma.schemas.ObservationDB.Enums.EditType.Created
+import lucuma.schemas.ObservationDB.Enums.EditType.Updated
```
Comment (rpiaggio) [inline common/src/main/scala/explore/cache/ProgramCache.scala:28]: Code seems to use `EditType.Created`and `EditType.Updated`. Maybe these 2 imports can be omitted?

**Review by rpiaggio — APPROVED** [review]
LGTM, with a couple of very minor comments.

File: common/src/main/scala/explore/cache/ProgramCache.scala (hunk 1 above)
Comment (hugo-vrijswijk) [reply common/src/main/scala/explore/cache/ProgramCache.scala]: It was used in a pattern match, but that's now matching on EditType.Created/Updated too

File: explore/src/main/scala/explore/observationtree/ObsList.scala [hunk 2]
```diff
@@ -52,14 +54,14 @@ case class ObsList(
   focusedObs:      Option[Observation.Id],
   focusedTarget:   Option[Target.Id],
   setSummaryPanel: Callback,
-  groups:          Pot[View[List[AllGroupElements]]]
+  groups:          GroupList
```
Comment (cquiroz) [inline explore/src/main/scala/explore/observationtree/ObsList.scala:57]: Much nicer :D

**Review by cquiroz — APPROVED** [review]
LGTM

**Pull request merged**